### DISCUSSION TITLE
Ship guns disappear (get reset to bases) upon entering "The End" (final mission)

### DISCUSSION THREAD

**Discussion by CorruptedPants** [FakeFishGames/Barotrauma#13681] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Upon entering the final mission of the game, "The End", where you fight Jo-wey, we were horrified to learn that all of our ship weapons had disappeared. We have completed "The End" several times before and this has never happened before. The game, as far as I am aware, does not have a mechanic to remove weapons.

### Reproduction steps

1. Load save game of us entering "The End"
2. Confirm there are no ship weapons anymore

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.2.8.0 (Winter Update hotfix 2)

### Which operating system did you encounter this bug on?

Linux

### Relevant error messages and crash reports

```shell
We are using a custom sub with an attached autonomous shuttle.
```

**Comment by CorruptedPants** [comment]
[save-file.zip](https://github.com/Regalis11/Barotrauma/files/14012430/save-file.zip)
Added save file

**Comment by giists** [comment]
I opened the save file and the guns are indeed gone, hardpoints are still there though. Unfortunately this doesn't really help with reproducing the problem, just confirming the bug already happened. If someone encounters this again, could you send a save from before entering the end?

**Comment by Regalis11** [comment]
>  could you send a save from before entering the end?

Finding one might be hard, it doesn't seem likely anyone would make a copy of the save _before_ the bug occurs. Since it seems like the turrets are not completely disappearing, but getting replaced with hardpoints, we could attempt to diagnose this though and see why they're getting replaced.

**Comment by Regalis11** [comment]
Giving the save file a test, it seems swapping the turrets to hardpoints is pending, which would normally imply that someone opted to swap them the last round. I'm not sure how that's possible in this case though. :/ Let's keep the ticket open and hope we can find some more clues for diagnosing this.

**Comment by CorruptedPants** [comment]
I have a suspicion that it is related to the ship having a drone. I ran a follow up campaign on a ship without a drone (rather, the same ship, but I got rid of the drone before entering The End) and the guns did not reset to base. I am currently running a modded campaign with a different ship (that also has a drone) and will keep around backup save files from before entering The End.

**Comment by CorruptedPants** [comment]
Did finish and guns did not disappear, can provide save files on request but as this instance did not reproduce the issue, I'll only do so on request
